src/App.tsx

```diff
@@ -22,19 +22,18 @@ const App: React.FC = () => {
 
   useEffect(() => {
     const updateUser = async () => {
-      try {
-        const cognitoUser = await Auth.currentAuthenticatedUser();
-        const { name, email } = await cognitoUser.attributes;
-        dispatch(setUser({ name, email }));
-      } catch (e) {
-        dispatch(setUser(emptyUser));
-      }
+      dispatch(getCurrentAuthenticatedUser());
     };
     Hub.listen('auth', updateUser);
     updateUser();
     return () => Hub.remove('auth', updateUser);
   }, [dispatch]);
 
+  useEffect(() => {
```

Comment:
hvorfor to useEffects?

Reply:
det er en fiffig bug i aws-amplify som gjør at hvis du tar inn data i den samme funksjonen som du skal gjøre et kall, så blir token-endepunktet kalt to ganger :))
[https://github.com/aws-amplify/amplify-js/3642]